validator/client/beacon-api/json_rest_handler.go

```diff
@@ -13,6 +14,7 @@ import (
 
 type jsonRestHandler interface {
 	GetRestJsonResponse(query string, responseJson interface{}) (*apimiddleware.DefaultErrorJson, error)
+	PostRestJson(apiEndpoint string, headers map[string]string, data *bytes.Buffer, responseJson interface{}) (*apimiddleware.DefaultErrorJson, error)
```

Comment:
I'm also not a fan of returning a value through an input parameter, but I'm not sure how to make this work. The unmarshal into the `interface{}` does work, but the struct type assert fails because the type is actually a `map`, not a struct with json tags.

Reply:
A map? I see you passing either `nil` or `rpcmiddleware.GenesisResponseJson{}` (in a test).

Reply:
Since we use `nil` in production code, it makes me think that this parameter should be removed. But I understand it will become useful later.

Reply:
> A map? I see you passing either `nil` or `rpcmiddleware.GenesisResponseJson{}` (in a test).

I meant that if we return an `interface{}` instead of passing it as a `responseJson` parameter, the unmarshalld data will be a map because the unmarshaller won't have any idea what the struct looks like.